tests/types/test-array.py

```diff
@@ -114,3 +114,79 @@ def test_create_from_arrow(pa):
     assert DT.type == dt.Type.arr32(dt.Type.int32)
     assert DT.names == ("A",)
     assert DT.to_list() == [src]
+
+
+def test_create_from_python1():
+    src = [[1, 2, 3], [], [4, 5], [6], None, [7, 8, 10, -1]]
+    DT = dt.Frame(A=src)
+    assert DT.shape == (6, 1)
+    assert DT.type == dt.Type.arr32(dt.Type.int32)
+    assert DT.names == ("A",)
+    assert DT.to_list() == [src]
+
+
+def test_create_from_python2():
+    src = [None, [1.5, 2, 3], [], None, [7, 8.99, 10, None, -1]]
+    DT = dt.Frame(B=src)
+    assert DT.shape == (5, 1)
+    assert DT.type == dt.Type.arr32(dt.Type.float64)
+    assert DT.names == ("B",)
+    assert DT.to_list() == [src]
+
+
+def test_create_from_python3():
+    src = [[], [], [], []]
+    DT = dt.Frame(D=src)
+    assert DT.shape == (4, 1)
+    assert DT.type == dt.Type.arr32(dt.Type.void)
+    assert DT.names == ("D",)
+    assert DT.to_list() == [src]
+
+
+def test_create_from_python4():
+    src = [None, [], [], [None] * 5, [], None, []]
+    DT = dt.Frame(E=src)
+    assert DT.shape == (7, 1)
+    assert DT.type == dt.Type.arr32(dt.Type.void)
+    assert DT.names == ("E",)
+    assert DT.to_list() == [src]
+
+
+def test_create_from_python5():
+    src = [["a", "b", "c"], None, ["d"], ["efg", None]]
+    DT = dt.Frame(F=src)
+    assert DT.shape == (4, 1)
+    assert DT.type == dt.Type.arr32(dt.Type.str32)
+    assert DT.names == ("F",)
+    assert DT.to_list() == [src]
+
+
+def test_create_from_python_array_of_arrays():
+    src = [
+            [[1], [2, 3]],
+            [],
+            None,
+            [[0, 4, 111], None, [15, -2]]
+          ]
+    DT = dt.Frame(N=src)
+    assert DT.shape == (4, 1)
+    assert DT.type == dt.Type.arr32(dt.Type.arr32(dt.Type.int32))
+    assert DT.names == ("N",)
+    assert DT.to_list() == [src]
+
+
+def test_create_from_python_nested():
+    src = [[[[[[]]]]]]
+    DT = dt.Frame(Q=src)
+    a = dt.Type.arr32
+    assert DT.shape == (1, 1)
+    assert DT.type == a(a(a(a(a(dt.Type.void)))))
+    assert DT.to_list() == [src]
+
+
+def test_create_from_python_array_incompatible_child_types():
+    src = [[1, 2, 3], ["one"]]
+    msg = "Cannot create column: element at index 3 is of type " \
+          "<class 'str'>, whereas previous elements were int32"
+    with pytest.raises(TypeError, match=msg):
+        DT = dt.Frame(E=src)
```

Comment:
What about adding a test that will modify the original list and then will make sure the frame's content didn't change?

Reply:
Since there is not even a theoretical way for the python list of objects to affect the content of a datatable's Column, I don't think there is a need for such a test.